### DISCUSSION TITLE
Issue with integrating PocketBase routes under custom path using Gorilla mux

### DISCUSSION THREAD

**Discussion by Mohamedjcali** [pocketbase/pocketbase#5707] [Q&A]
Hello, I'm working on integrating PocketBase as part of an existing Go web server. I already have several custom APIs running under a Gorilla mux router. The challenge I'm facing is getting PocketBase to serve its API under a custom path (/db/) while keeping my existing routes working. I need PocketBase to respond under the path http://127.0.0.1:8080/db/ (e.g., http://127.0.0.1:8080/db/api/admin to create an admin), but the PocketBase routes are returning a 404 error even though PocketBase seems to be initializing (as the pb_data folder is being created).

Here is my code:

```go
package main

import (
	"fmt"
	"log"
	"net/http"
	"tfen/user"

	"github.com/gorilla/mux"
	"github.com/pocketbase/pocketbase"
	"github.com.pocketbase/pocketbase/core"
)

func HomeHandler(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintln(w, "Welcome to the home page!")
}

func main() {
	// Initialize PocketBase
	pb := pocketbase.New()

	
	router := mux.NewRouter()

	
	router.HandleFunc("/api/signup", user.Signup).Methods("POST")
	router.HandleFunc("/", HomeHandler).Methods("GET")
	router.HandleFunc("/api/login", user.Login).Methods("POST")
	router.HandleFunc("/api/login/google", user.GoogleLoginURL).Methods("GET")
	router.HandleFunc("/callback", user.GoogleCallback).Methods("GET")
	router.HandleFunc("/api/get-user/{user-id}", user.GetUserByToken).Methods("GET")
	router.HandleFunc("/api/create-project/{user_id}", user.CreateProject).Methods("POST")
	router.HandleFunc("/ws", user.HandleWebSocket)

	
	pb.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		// Mount PocketBase routes under `/db/`
		router.PathPrefix("/db/").Handler(http.StripPrefix("/db", e.Router))
		return nil
	})

	
	go func() {
		if err := pb.Start(); err != nil {
			log.Fatal(err)
		}
	}()

	
	log.Fatal(http.ListenAndServe(":8080", router))
}
```
## Issue details:

PocketBase seems to start correctly because the pb_data folder is created.
When I try to hit the PocketBase API endpoint http://127.0.0.1:8080/db/api/collections using Postman, I receive a 404 page not found error.
I can still access my custom routes such as http://127.0.0.1:8080/api/signup, but PocketBase routes do not seem to work under /db/.
Questions:

Is this the correct approach for mounting PocketBase routes under /db/ while keeping my custom APIs under other paths?
How can I ensure PocketBase is correctly mounted under /db/ so that I can make requests like POST http://127.0.0.1:8080/db/api/admin without getting a 404 error?
Can I configure PocketBase to save the pb_data folder to a custom location (e.g., the Desktop folder) and rename it to something like db instead of pb_data?
## Things I've tried:

Changing the route prefix for PocketBase using ```http.StripPrefix("/db", e.Router)```
 to ensure PocketBase routes are prefixed with /db/.
Checking if PocketBase is running correctly by confirming the creation of the pb_data folder, which suggests the initialization works but routing might be the issue.
## Desired behavior:

I want PocketBase to be served under http://127.0.0.1:8080/db/ so that I can access its APIs at paths like http://127.0.0.1:8080/db/api/admin, while my custom routes still work as expected.
I also want to change the default storage folder from pb_data to db and move it to a custom directory.
Any guidance on how to resolve this issue would be appreciated!

**Comment by whmeitzler** [comment]
I think you're signing up for more pain than you expect by trying to move PB to a subpath. 
Wiring together tow different routers, while (obviously) possible often leads to odd behavior due to  the differing idiosyncrasies of each router implementation. Furthermore, moving the Pocketbase APIS to a sub-path will (likely) break the JS client lib, and therefore the admin UI. 

I would try flipping the script- integrating your Gorilla endpoints to Pocketbase, rather than the other way round.


Regarding changing the storage dir, there is a command line switch (`--dev`) for that

**Reply by dschissler** [reply]
> Furthermore, moving the Pocketbase APIS to a sub-path will (likely) break the JS client lib, and therefore the admin UI.

You would just have to pass in a path to the js client instead of relying on the default.

```javascript
    constructor(baseUrl = "/", authStore?: BaseAuthStore | null, lang = "en-US") {
```
